code/modules/mining/boulder_processing/_boulder_processing.dm

```diff
@@ -0,0 +1,355 @@
+/obj/machinery/bouldertech
+	name = "bouldertech brand refining machine"
+	desc = "You shouldn't be seeing this! And bouldertech isn't even a real company!"
+	icon = 'icons/obj/machines/mining_machines.dmi'
+	icon_state = "ore_redemption"
+	anchored = TRUE
+	density = TRUE
+
+	/// What is the efficiency of minerals produced by the machine?
+	var/refining_efficiency = 1
+	/// How many boulders can we process maximum per loop?
+	var/boulders_processing_max = 1
+	/// What boulder(s) are we holding?
+	var/list/boulders_contained = list()
+	/// How many boulders can we hold maximum?
+	var/boulders_held_max = 1
+	/// Does this machine have a mineral storage link to the silo?
+	var/holds_minerals = FALSE
+	/// What materials do we accept and process out of boulders? Removing iron from an iron/glass boulder would leave a boulder with glass.
+	var/list/processable_materials = list()
+	/// If we get a boulder with one of these inside, we'll drop it back out.
+	var/static/list/drop_if_contained = list(
+		/obj/item/relic,
+	)
```

Comment:
if this list is static and only contains 1 element then why even make it a list.? Just write an `istype()` check there directly.

Reply:
I do agree conceptually, however the relic boulder framework is set up to be able to contain more things going forward. Yes, right NOW it's just one item (as an example more than anything), but I feel leaving this here for now is helpful for expandability.